aten/src/ATen/native/Sorting.cpp

```diff
@@ -14,30 +14,68 @@
 
 namespace at {
 namespace meta {
+
 using namespace native;
-  TORCH_META_FUNC(topk) (
-    const Tensor& self,
-    int64_t k,
-    int64_t dim_,
-    bool largest,
-    bool sorted) {
 
-    int64_t dim = maybe_wrap_dim(dim_, self.dim(), /*wrap_scalar=*/true);
-    TORCH_CHECK(
-        k >= 0 && k <= (self.dim() > 0 ? self.size(dim) : 1),
-        "selected index k out of range");
-    int64_t sliceSize = self.dim() == 0 ? 1 : self.size(dim);
-    TORCH_CHECK(k >= 0 && k <= sliceSize, "k not in range for dimension");
-
-    // Build the output size, which is the dim being selected set to
-    // size k
-    DimVector topKSize(self.sizes().vec());
-    if (topKSize.size() > 0) {
-      topKSize[dim] = k;
-    }
-    set_output(0, topKSize, self.options());
-    set_output(1, topKSize, self.options().dtype(at::kLong));
+TORCH_META_FUNC(topk)
+(const Tensor& self, int64_t k, int64_t dim_, bool largest, bool sorted) {
+  int64_t dim = maybe_wrap_dim(dim_, self.dim(), /*wrap_scalar=*/true);
+  TORCH_CHECK(
+      k >= 0 && k <= (self.dim() > 0 ? self.size(dim) : 1),
+      "selected index k out of range");
+  int64_t sliceSize = self.dim() == 0 ? 1 : self.size(dim);
+  TORCH_CHECK(k >= 0 && k <= sliceSize, "k not in range for dimension");
+
+  // Build the output size, which is the dim being selected set to
+  // size k
+  DimVector topKSize(self.sizes().vec());
+  if (topKSize.size() > 0) {
+    topKSize[dim] = k;
+  }
+  set_output(0, topKSize, self.options());
+  set_output(1, topKSize, self.options().dtype(at::kLong));
+}
+
+TORCH_META_FUNC2(sort, stable)
+(const Tensor& self, c10::optional<bool> stable, int64_t dim, bool descending) {
+  TORCH_INTERNAL_ASSERT(
+      stable.has_value(),
+      "sort(): c10::optional<bool> for stable has to have value.");
+  maybe_wrap_dim(dim, self.dim());
+
+  const auto& values = maybe_get_output(0);
+  auto values_defined = values.defined();
+  const auto& indices = maybe_get_output(1);
+  auto indices_defined = indices.defined();
+
+  std::vector<int64_t> strides = (self.is_non_overlapping_and_dense())
+      ? self.strides().vec()
+      : at::infer_dense_strides(self.sizes(), self.strides());
+
+  // 'set_output' will only modify the strides, if the output was resized.
+  // Which means that we can ignore the output's strides here.
+  set_output(0, self.sizes(), strides, self.options(), {});
+  set_output(1, self.sizes(), strides, self.options().dtype(kLong), {});
+
+  // Overlap Checking
+  // 1. 'self' can be the same as 'values' (in-place).
+  // 2. 'self' can't overlap with 'indices'
+  // 3. 'values' and 'indices' can't overlap.
+  if (values_defined) {
+    at::assert_no_internal_overlap(values);
+    at::assert_no_partial_overlap(self, values);
   }
+
+  if (indices_defined) {
+    at::assert_no_internal_overlap(indices);
+    at::assert_no_overlap(self, indices);
+  }
+
+  if (values_defined && indices_defined) {
+    at::assert_no_overlap(values, indices);
+  }
```

Comment:
Do you think these overlap checks make sense? The `sort` kernel didn't really check for them before, but I thought I would introduce, anyway.

Reply:
oh, thanks for calling that out.

That seems like a pretty reasonable check, since otherwise you'd get silently incorrect results.

I would still make it a separate PR though, that way there aren't any unnecessarily "functionality changes" in the port itself. Do you mind making it a separate PR after this one in the stack